src/main/java/seedu/address/model/attendance/Week.java

```diff
@@ -0,0 +1,57 @@
+package seedu.address.model.attendance;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.util.Objects;
+
+import seedu.address.model.lesson.NumberOfOccurrences;
+
+/**
+ * Represents a week number.
+ * Guarantees: immutable; is valid as declared in {@link #isValidWeek(int)}
+ */
+public class Week {
+
+    public static final int LOWER_BOUND = NumberOfOccurrences.LOWER_BOUND;
+    public static final int UPPER_BOUND = NumberOfOccurrences.UPPER_BOUND;
+    public static final String MESSAGE_CONSTRAINTS =
+            String.format("The week number must be a number between %d to %d.", LOWER_BOUND, UPPER_BOUND);
+    private final int week;
+
+    /**
+     * Constructor method.
+     * Checks that week is valid.
+     */
+    public Week(int week) {
+        checkArgument(isValidWeek(week), MESSAGE_CONSTRAINTS);
+
+        this.week = week;
+    }
+
+    /**
+     * Returns true if the given week is an integer between 1 and 50.
+     */
+    public static boolean isValidWeek(int week) {
+        return LOWER_BOUND <= week && week <= UPPER_BOUND;
+    }
+
+    public int getWeekNumber() {
+        return week;
+    }
+
+    public int getZeroBasedWeekNumber() {
+        return week - 1;
+    }
+
```

Comment:
Instead of introducing this method within the model, perhaps it would be better for the methods that require a zero based number to use the currently available `Index` object and retrieve a `getOneBased()` value instead?

That is, the method that does the comparison should be the one performing the conversion from a zero based index to a one based index for comparison with the `Week` value.